game/state/shared/agent.cpp

```diff
@@ -387,12 +389,67 @@ void Agent::assignTraining(TrainingAssignment assignment)
 
 void Agent::hire(GameState &state, StateRef<Building> newHome)
 {
+	if (owner == state.getPlayer())
+	{
+		LogError("Agent is already owned by x-com");
+	}
+
+	oldOwner = owner;
 	owner = newHome->owner;
 	homeBuilding = newHome;
 	recentlyHired = true;
 	setMission(state, AgentMission::gotoBuilding(state, *this, newHome, false, true));
 }
 
+void Agent::fire(GameState &state)
+{
+	if (owner != state.getPlayer())
+	{
+		LogError("Cannot fire agents from other organizations");
+	}
+
+	if (currentVehicle)
+	{
+		LogError("Should not be able to fire people inside a vehicle");
+	}
+
+	removeFromLab(state);
+
+	if (oldOwner == owner)
+	{
+		// If agent home org is x-com default new owner to government (for starting agents)
+		owner = state.getGovernment();
+	}
+	else
+	{
+		owner = oldOwner;
+	}
+
+	if (!owner->buildings.empty())
+	{
+		homeBuilding = owner->buildings.front();
```

Comment:
Is this correct? If a org has multiple buildings, all fired agents will return to the first one in the list?

What does the OG do? Random of each building? Store which building they were initially assigned? Just magically make the agent disappear as the "home building" of agents is a mechanic with little gameplay use?

Reply:
Yes that is how it works currently. 
I am not sure what happens in OG to be honest. But since this has almost no effect on gameplay I thought we could keep it simple. The scientists just travel between some building and x-com base when hired or fired.

There might be a problem with this solution though if there is no buildings available for the organization.  If you have better idea how to handle this I can make changes.